File: apps/web/src/types/traveller.ts
export interface TravellerIdentity {
  name: string;
  email: string;
  locale: string;
  timezone: string;
}

export interface TravellerPreferences {
  seat: "window" | "aisle" | "no_preference";
  cabin_class: "economy" | "business" | "first";
  meal: string;
  accommodation_type: string;
  budget_tier: "budget" | "mid" | "luxury";
}

export interface AirlineLoyalty {
  carrier: string;
  number: string;
}

export interface HotelLoyalty {
  brand: string;
  number: string;
}

export interface TravellerLoyalty {
  airline_programs: AirlineLoyalty[];
  hotel_programs: HotelLoyalty[];
}

export interface TravellerProfile {
  id: string;
  created_at: string;
  updated_at: string;
  identity: TravellerIdentity;
  preferences: TravellerPreferences;
  loyalty: TravellerLoyalty;
  travel_history: unknown[];
}

export interface CreateProfileRequest {
  identity: TravellerIdentity;
  preferences: TravellerPreferences;
  loyalty: TravellerLoyalty;
}

File: services/api/app/models/traveller.py
from pydantic import BaseModel


class IdentitySchema(BaseModel):
    name: str
    email: str
    locale: str = "en"
    timezone: str = "UTC"


class PreferencesSchema(BaseModel):
    seat: str = "no_preference"
    cabin_class: str = "economy"
    meal: str = "standard"
    accommodation_type: str = "hotel"
    budget_tier: str = "mid"


class AirlineLoyaltySchema(BaseModel):
    carrier: str
    number: str


class HotelLoyaltySchema(BaseModel):
    brand: str
    number: str


class LoyaltySchema(BaseModel):
    airline_programs: list[AirlineLoyaltySchema] = []
    hotel_programs: list[HotelLoyaltySchema] = []


class CreateProfileRequest(BaseModel):
    identity: IdentitySchema
    preferences: PreferencesSchema = PreferencesSchema()
    loyalty: LoyaltySchema = LoyaltySchema()


class TravellerProfileResponse(BaseModel):
    id: str
    created_at: str
    updated_at: str
    identity: IdentitySchema
    preferences: PreferencesSchema
    loyalty: LoyaltySchema
    travel_history: list = []
